### PR TITLE
Add azure storage

### PR DESCRIPTION
**Pull request by victorursan**

### CODE REVIEW

**Comment by victorursan** [comment]
@tethryus waiting on cs-azure:0.0.4 😄

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/delete_container.sl [hunk 1]
```diff
@@ -7,148 +7,77 @@
 #
 ########################################################################################################################
 #!!
-#! @description: Performs an HTTP request to delete a container
+#! @description: Create a container for the specified account
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/delete_container.sl]: @victorursan Delete

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/delete_container.sl [hunk 2]
```diff
@@ -7,148 +7,77 @@
...
-    - auth_token
-    - api_version:
-        required: false
-        default: '2015-04-05'
-    - list_cont_auth_header
-    - date
-    - storage_account
+    - account_name
+    - accountName:
+        default: ${get("account_name", ""}
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/delete_container.sl]: required: false

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/delete_container.sl [hunk 3]
```diff
@@ -7,148 +7,77 @@
...
+        private: true
+    - key:
+        sensitive: true
     - container_name
+    - containerName:
+        default: ${get("container_name", ""}
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/delete_container.sl]: required: false

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl [hunk 4]
```diff
@@ -7,145 +7,77 @@
...
 #!
-#! @input list_cont_auth_header: Azure Storage authorization key
-#! @input api_version: The API version used to create calls to Azure Storage
-#!                     Default: '2015-04-05'
-#! @input date: Specifies the Coordinated Universal Time (UTC) for the request
-#! @input storage_account: Storage account name
-#! @input container_name: Container name
+#! @input account_name: Azure account name
+#! @input key: Azure account key
+#! @input container_name: the name you want to give the new container
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl]: to the

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl [hunk 5]
```diff
@@ -7,145 +7,77 @@
...
   inputs:
-    - list_cont_auth_header
-    - api_version:
-        required: false
-        default: '2015-04-05'
-    - date
-    - storage_account
+    - account_name
+    - accountName:
+        default: ${get("account_name", ""}
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl]: required: false

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl [hunk 6]
```diff
@@ -7,145 +7,77 @@
...
+        private: true
+    - key:
+        sensitive: true
     - container_name
+    - containerName:
+        default: ${get("container_name", ""}
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl]: required: false

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_containers.sl [hunk 7]
```diff
@@ -7,147 +7,72 @@
...
-    - auth_token
-    - api_version:
-        required: false
-        default: '2015-04-05'
-    - list_cont_auth_header
-    - date
-    - storage_account
+    - account_name
+    - accountName:
+        default: ${get("account_name", ""}
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_containers.sl]: required: false

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/create_container.sl [hunk 8]
```diff
@@ -47,109 +28,56 @@
...
-    - auth_token
-    - api_version:
-        required: false
-        default: '2015-04-05'
-    - list_cont_auth_header
-    - date
-    - storage_account
+    - account_name
+    - accountName:
+        default: ${get("account_name", ""}
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/create_container.sl]: required: false

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/create_container.sl [hunk 9]
```diff
@@ -47,109 +28,56 @@
...
+        private: true
+    - key:
+        sensitive: true
     - container_name
+    - containerName:
+        default: ${get("container_name", ""}
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/create_container.sl]: required: false

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/delete_blob.sl [hunk 10]
```diff
@@ -7,176 +7,83 @@
 #
 ########################################################################################################################
 #!!
-#! @description: Performs an HTTP request to delete a blob inside a container
+#! @description: Delete a container from the specified account
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/delete_blob.sl]: Delete a blob from the container that exists in the specified storage account.

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/delete_container.sl (hunk 2 above)
Comment (victorursan) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/delete_container.sl]: :+1:

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl (hunk 4 above)
Comment (victorursan) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl]: :+1:

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl (hunk 5 above)
Comment (victorursan) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl]: :+1:

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl (hunk 6 above)
Comment (victorursan) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl]: :+1:

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_containers.sl (hunk 7 above)
Comment (victorursan) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_containers.sl]: :+1:

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/create_container.sl (hunk 8 above)
Comment (victorursan) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/create_container.sl]: :+1:

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/create_container.sl (hunk 9 above)
Comment (victorursan) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/create_container.sl]: :+1:

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/delete_blob.sl (hunk 10 above)
Comment (victorursan) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/delete_blob.sl]: :+1:

**Comment by victorursan** [comment]
Can I do something like : 
"
- storage_account
- accountName:
        default: ${get("storage_account", ""}
        required: false
        private: true 
"
or is it not a good "practice" ? :))

**Comment by tethryus** [comment]
You could and then in the next release we could rename the camelCase value which won`t impact the operation or any flow that uses it.

**Comment by victorursan** [comment]
I think it's better to postpone the release (cs-azure:0.04) until I modify the input in the operation and you can review it and merge it and after that release it ...

**Comment by tethryus** [comment]
@victorursan Agreed, Other than that great work 👍

**Comment by victorursan** [comment]
:+1:

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_containers.sl [hunk 11]
```diff
@@ -7,147 +7,73 @@
...
-#! @output status_code: 200 if request completed successfully, others in case something went wrong
-#! @output error_message: If the containers are not found the error message will be populated with a response,
-#!                        empty otherwise
+#! @output output: the container name that was created
+#! @output return_code: 0 if request completed successfully, -1 in case something went wrong
+#! @output exception: the stacktrace of the operation in case something went wrong
 #!
-#! @result SUCCESS: The list of the containers under the specified account retrieved successfully.
-#! @result FAILURE: There was an error while trying to retrieve the list of the containers under the specified account
+#! @result SUCCESS: Container created successfully.
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_containers.sl]: Container list retrieved successfully.

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_containers.sl [hunk 12]
```diff
@@ -7,147 +7,73 @@
...
+#! @result FAILURE: There was an error while trying to create the container.
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_containers.sl]: to list the containers

File: content/io/cloudslang/microsoft/azure/utility/get_auth_token.sl [hunk 13]
```diff
@@ -45,9 +45,12 @@ operation:
...
         private: true
-    - authority:
+    - login_authority:
         default: 'https://sts.windows.net/common'
         required: false
+    - loginAuthority:
+        default: ${get("login_authority", "")}
+        required: false
         private: true
     - resource:
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/utility/get_auth_token.sl]: Please remove the private tag of resource

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl [hunk 14]
```diff
@@ -7,145 +7,79 @@
...
-#! @output status_code: 200 if request completed successfully, others in case something went wrong
-#! @output error_message: If the containers are not found the error message will be populated with a response,
-#!                        empty otherwise
+#! @output output: the container name that was created
+#! @output return_code: 0 if request completed successfully, -1 in case something went wrong
+#! @output exception: the stacktrace of the operation in case something went wrong
 #!
-#! @result SUCCESS: The list of blobs under the specified container retrieved successfully.
-#! @result FAILURE: There was an error while trying to retrieve the list of blobs under the specified container
+#! @result SUCCESS: Container created successfully.
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl]: The list of blobs under the specified container retrieved successfully.

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl [hunk 15]
```diff
@@ -7,145 +7,79 @@
 #
 ########################################################################################################################
 #!!
-#! @description: Performs an HTTP request to retrieve a list of the blobs under the specified container
+#! @description: List the blobs for the specified account
```
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_blobs.sl]: storage account

File: content/io/cloudslang/microsoft/azure/compute/storage/containers/list_containers.sl (hunk 11 above)
Comment (tethryus) [inline content/io/cloudslang/microsoft/azure/compute/storage/containers/list_containers.sl]: @result SUCCESS: The list of the containers under the specified account retrieved successfully.
@result FAILURE: There was an error while trying to retrieve the list of the containers under the specified account

**Pull request merged**